File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IVideoApi.java
package dev.ragnarok.fenrir.api.interfaces;

import androidx.annotation.CheckResult;

import java.util.Collection;

import dev.ragnarok.fenrir.api.model.AccessIdPair;
import dev.ragnarok.fenrir.api.model.IAttachmentToken;
import dev.ragnarok.fenrir.api.model.Items;
import dev.ragnarok.fenrir.api.model.VKApiVideo;
import dev.ragnarok.fenrir.api.model.VKApiVideoAlbum;
import dev.ragnarok.fenrir.api.model.response.DefaultCommentsResponse;
import dev.ragnarok.fenrir.api.model.response.SearchVideoResponse;
import io.reactivex.rxjava3.core.Single;


public interface IVideoApi {

    @CheckResult
    Single<DefaultCommentsResponse> getComments(Integer ownerId, int videoId, Boolean needLikes,
                                                Integer startCommentId, Integer offset, Integer count, String sort,
                                                Boolean extended, String fields);

    @CheckResult
    Single<Integer> addVideo(Integer targetId, Integer videoId, Integer ownerId);

    @CheckResult
    Single<Integer> deleteVideo(Integer videoId, Integer ownerId, Integer targetId);

    @CheckResult
    Single<Items<VKApiVideoAlbum>> getAlbums(Integer ownerId, Integer offset, Integer count, Boolean needSystem);

    @CheckResult
    Single<Items<VKApiVideoAlbum>> getAlbumsByVideo(Integer target_id, Integer owner_id, Integer video_id);

    @CheckResult
    Single<SearchVideoResponse> search(String query, Integer sort, Boolean hd, Boolean adult, String filters,
                                       Boolean searchOwn, Integer offset, Integer longer, Integer shorter,
                                       Integer count, Boolean extended);

    @CheckResult
    Single<Boolean> restoreComment(Integer ownerId, int commentId);

    @CheckResult
    Single<Boolean> deleteComment(Integer ownerId, int commentId);

    @CheckResult
    Single<Items<VKApiVideo>> get(Integer ownerId, Collection<AccessIdPair> ids, Integer albumId,
                                  Integer count, Integer offset, Boolean extended);

    @CheckResult
    Single<Integer> createComment(Integer ownerId, int videoId, String message,
                                  Collection<IAttachmentToken> attachments, Boolean fromGroup,
                                  Integer replyToComment, Integer stickerId, Integer uniqueGeneratedId);


    @CheckResult
    Single<Boolean> editComment(Integer ownerId, int commentId, String message, Collection<IAttachmentToken> attachments);

    @CheckResult
    Single<Boolean> edit(Integer ownerId, int video_id, String name, String desc);

}

File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IVideoApi.kt
package dev.ragnarok.fenrir.api.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.api.model.*
import dev.ragnarok.fenrir.api.model.response.DefaultCommentsResponse
import dev.ragnarok.fenrir.api.model.response.SearchVideoResponse
import io.reactivex.rxjava3.core.Single

interface IVideoApi {
    @CheckResult
    fun getComments(
        ownerId: Int?, videoId: Int, needLikes: Boolean?,
        startCommentId: Int?, offset: Int?, count: Int?, sort: String?,
        extended: Boolean?, fields: String?
    ): Single<DefaultCommentsResponse>

    @CheckResult
    fun addVideo(targetId: Int?, videoId: Int?, ownerId: Int?): Single<Int>

    @CheckResult
    fun deleteVideo(videoId: Int?, ownerId: Int?, targetId: Int?): Single<Int>

    @CheckResult
    fun getAlbums(
        ownerId: Int?,
        offset: Int?,
        count: Int?,
        needSystem: Boolean?
    ): Single<Items<VKApiVideoAlbum>>

    @CheckResult
    fun getAlbumsByVideo(
        target_id: Int?,
        owner_id: Int?,
        video_id: Int?
    ): Single<Items<VKApiVideoAlbum>>

    @CheckResult
    fun search(
        query: String?, sort: Int?, hd: Boolean?, adult: Boolean?, filters: String?,
        searchOwn: Boolean?, offset: Int?, longer: Int?, shorter: Int?,
        count: Int?, extended: Boolean?
    ): Single<SearchVideoResponse>

    @CheckResult
    fun restoreComment(ownerId: Int?, commentId: Int): Single<Boolean>

    @CheckResult
    fun deleteComment(ownerId: Int?, commentId: Int): Single<Boolean>

    @CheckResult
    operator fun get(
        ownerId: Int?, ids: Collection<AccessIdPair>?, albumId: Int?,
        count: Int?, offset: Int?, extended: Boolean?
    ): Single<Items<VKApiVideo>>

    @CheckResult
    fun createComment(
        ownerId: Int, videoId: Int, message: String?,
        attachments: Collection<IAttachmentToken>?, fromGroup: Boolean?,
        replyToComment: Int?, stickerId: Int?, uniqueGeneratedId: Int?
    ): Single<Int>

    @CheckResult
    fun editComment(
        ownerId: Int,
        commentId: Int,
        message: String?,
        attachments: Collection<IAttachmentToken>?
    ): Single<Boolean>

    @CheckResult
    fun edit(ownerId: Int, video_id: Int, name: String?, desc: String?): Single<Boolean>
}